### ISSUE TITLE
[docs] Fix demos acting flaky in Argos

### ISSUE THREAD

**Issue by LukasTy** [mui/mui-x#15372]
Some demos are producing flaky results in Argos.
Sometimes we approve the broken versions and then have to re-approve the proper version of those snapshots in multiple follow-up PRs.
Here are examples of the latest demos producing issues: https://app.argos-ci.com/mui/mui-x/builds/26329/118165897, https://app.argos-ci.com/mui/mui-x/builds/26330/118217890.

- [ColumnSpanningDerived](https://mui.com/x/react-data-grid/column-spanning/#function-signature:~:text=Function%20signature%20can%20also%20be%20useful%20to%20derive%20colSpan%20value%20from%20row%20data%3A)
- [PopularFeaturesDemo](https://mui.com/x/react-data-grid/demo/)
- [RowSpanningCalendar](https://mui.com/x/react-data-grid/row-spanning/#demo)
- ColumnSelectorGridSnap

**Comment by JCQuintas** [comment]
Could this be an issue with the `v7` branch and the config? I've run the migration scrips locally in master and v7 and the outputs are different for some of these tests

**Comment by LukasTy** [comment]
As far as I remember, at least he `PopularFeaturesDemo` was flaky before the start of the `master`<->`v7.x` "tango". 🤔

**Comment by JCQuintas** [comment]
Yeah, so from the argos config, it uses `master` as the base, so I GUESS if on a `v7` PR we accept the changes, they will be the new `"master"` changes.

**Comment by LukasTy** [comment]
> Yeah, so from the argos config, it uses `master` as the base, so I GUESS if on a `v7` PR we accept the changes, they will be the new `"master"` changes.

Hasn't Argos been semi recently updated to work smart with the branches? 🤔

**Comment by JCQuintas** [comment]
Idk, their settings doesn't have anything related to that: https://app.argos-ci.com/mui/mui-x/settings

**Comment by LukasTy** [comment]
> Idk, their settings doesn't have anything related to that: https://app.argos-ci.com/mui/mui-x/settings

@JCQuintas I was referring to https://argos-ci.com/changelog/2024-10-08-multi-branches. 😉

**Comment by oliviertassinari** [comment]
Related to #11163
